src/go/rc-conf/conf.go

```diff
@@ -0,0 +1,240 @@
+package rcconf
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"net"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/coreos/go-systemd/v22/dbus"
+)
+
+type Config struct {
+	locale           string
+	timezone         string
+	keymap           string
+	hostname         string
+	enabled_daemons  []string
+	disabled_daemons []string
+	netconfig        NetworkConfig
+}
+
+type NetworkConfig struct {
+	net_interface string
+	address       string
+	netmask       string
+	broadcast     string
+	gateway       string
+}
+
+func netmaskToCIDR(netmask string) (int, error) {
+	ip := net.ParseIP(netmask)
+	if ip == nil {
+		return 0, fmt.Errorf("invalid netmask: %s", netmask)
+	}
+
+	ip = ip.To4()
+	if ip == nil {
+		return 0, fmt.Errorf("invalid IPv4 address: %s", netmask)
+	}
+
+	cidr := 0
+	for _, b := range ip {
+		for b > 0 {
+			cidr += int(b & 1)
+			b >>= 1
+		}
+	}
+
+	return cidr, nil
+}
+
+func ParseConfig() Config {
+	config_file, ok := os.LookupEnv("RC_CONF")
+	if !ok {
+		config_file = "/etc/rc.conf"
+	}
+
+	file, err := os.Open(config_file)
+	if err != nil {
+		fmt.Fprintln(os.Stderr, err)
+	}
+	defer file.Close()
+
+	scanner := bufio.NewScanner(file)
+	cfg := Config{}
+	for scanner.Scan() {
+		line := scanner.Text()
+		if !strings.Contains(line, "=") || strings.HasPrefix(line, "#") {
+			continue
+		}
+
+		// Should all be well formed configs
+		parts := strings.Split(line, "=")
```

Comment:
Seems like [`strings.Cut`](https://pkg.go.dev/strings#Cut) is better suited for this task.